odoo/addons/base/models/res_users.py

```diff
@@ -786,6 +786,48 @@ def change_password(self, old_passwd, new_passwd):
         # use self.env.user here, because it has uid=SUPERUSER_ID
         return self.env.user.write({'password': new_passwd})
 
+    def _deactivate_portal_user(self):
+        """Try to remove the current portal user.
+
+        If it failed, archive the user instead and remove sensitive information.
+
+        This is used to give the opportunity to portal users to de-activate their accounts.
+        Indeed, as the portal users can easily create accounts, they will sometimes wish
+        it removed because they don't use this Odoo portal anymore.
+
+        Before this feature, they would have to contact the website or the support to get
+        their account removed, which could be tedious.
+        """
+        if any(not user.has_group('base.group_portal') for user in self):
+            raise UserError(_('Only the portal users can deactivate their accounts.'))
+
+        ip = request.httprequest.environ['REMOTE_ADDR'] if request else 'n/a'
+
+        users_deletion = []
+
+        for user in self:
+            log_context = f'Account deletion asked for "{user.login}" (#{user.id}) from {ip}.'
+            _logger.info('%s Archive the user and remove sensitive information.', log_context)
```

Comment:
How about including the current user in this?

"Account deletion requested from {current user} (ip: {ip}) for account: {deleted account}." 

(Also not sure why there is a "log_context" variable).